src/CommitmentsV2/SFA.DAS.CommitmentsV2.UnitTests/Services/OverlapCheckServiceTests.cs

```diff
@@ -0,0 +1,173 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Moq;
+using NUnit.Framework;
+using SFA.DAS.CommitmentsV2.Domain.Entities;
+using SFA.DAS.CommitmentsV2.Domain.Interfaces;
+using SFA.DAS.CommitmentsV2.Services;
+
+namespace SFA.DAS.CommitmentsV2.UnitTests.Services
+{
+    [TestFixture]
+    public class OverlapCheckServiceTests
+    {
+        private OverlapCheckServiceTestFixture _fixture;
+
+        [SetUp]
+        public void Arrange()
+        {
+            _fixture = new OverlapCheckServiceTestFixture();
+        }
+
+        [TestCase("2018-04-01", "2018-06-30", Description = "Start and end date both disregarded")]
+        [TestCase("2018-04-01", "2018-05-15", Description = "Start date disregarded")]
+        [TestCase("2018-05-15", "2018-06-01", Description = "End date disregarded")]
+
+        public async Task ThenTheOverlapCheckDisregardsDatesWithinTheSameMonth(DateTime startDate, DateTime endDate)
+        {
+            var result = await _fixture.WithDateRange(startDate, endDate).CheckForOverlaps();
+            Assert.IsFalse(result.HasOverlaps);
+        }
+
+        [TestCase("2017-01-01", "2017-12-31", Description = "Before any apprenticeships")]
+        [TestCase("2021-01-01", "2021-12-31", Description = "After any apprenticeships")]
+        public async Task ThenIfDatesDoNotFallWithinRangeOfExistingApprenticeshipThenNotOverlapping(DateTime startDate, DateTime endDate)
+        {
+            var result = await _fixture.WithDateRange(startDate, endDate).CheckForOverlaps();
+            Assert.IsFalse(result.HasOverlaps);
+        }
+
+        [Test]
+        public async Task ThenIfNoUlnsMatchInputThenNotOverlapping()
+        {
+            var result = await _fixture
+                .WithNoMatchingUlnUtilisations()
+                .WithDateRange(new DateTime(2018, 01, 1), new DateTime(2018, 12, 31))
+                .CheckForOverlaps();
+
+            Assert.IsFalse(result.HasOverlaps);
+        }
+
+        [Test]
+        public async Task ThenIfStartDateFallsWithinRangeOfExistingApprenticeshipThenIsOverlapping()
+        {
+            var result = await _fixture
+                .WithDateRange(new DateTime(2018, 03, 15), new DateTime(2018, 05, 15))
+                .CheckForOverlaps();
+
+            Assert.IsTrue(result.HasOverlappingStartDate);
+        }
+
+        [Test]
+        public async Task ThenIfEndDateFallsWithinRangeOfExistingApprenticeshipThenIsOverlapping()
+        {
+            var result = await _fixture
+                .WithDateRange(new DateTime(2018, 05, 15), new DateTime(2018, 07, 15))
+                .CheckForOverlaps();
+
+            Assert.IsTrue(result.HasOverlappingEndDate);
+        }
+
+        [TestCase("2018-03-01", "2018-03-31", Description = "Dates contained within existing range - single month")]
```

Comment:
Can this happen? Also the date range in of March doesn't overlap the test data (that I can see)